concent_api/core/payments/report.py

```diff
@@ -0,0 +1,759 @@
+        check_error_deposit_claim_exists_for_forced_payment(
+            errors,
+            forced_subtask_payment.tx_hash,
+        )
+
+    for settlement_payment in payments.settlement_payments:
+        check_error_deposit_claim_exists_for_settlement_payment(
+            errors,
+            settlement_payment.tx_hash,
+        )
+
+    return errors, warnings
+
+
+def check_additional_verification_payment_errors_and_warning(
+    errors: List[Error],
+    warnings: List[Warnings],
+    matched_objects: List[PendingResponse],
+    additional_verification_payment: CoverAdditionalVerificationEvent,
+) -> None:
+    """ Check errors and warning related to CoverAdditionalVerificationEvent object. """
+    error_was_added = check_error_cannot_match_payment_with_any_message(
+        errors,
+        matched_objects,
+        additional_verification_payment
+    )
+
+    if error_was_added:
+        check_warning_deposit_claim_exists_for_missing_match(
+            warnings,
+            additional_verification_payment.subtask_id,
+            ConcentUseCase.ADDITIONAL_VERIFICATION,
+        )
+
+    check_error_more_matching_message_for_payment_than_given_amount(
+        errors,
+        matched_objects,
+        additional_verification_payment,
+        amount=2,
+    )
+    check_error_more_than_one_subtask_result_of_the_same_type_for_subtask(
+        errors,
+        matched_objects,
+        additional_verification_payment,
+    )
+    check_warning_no_subtask_related_with_additional_verification_payment(
+        warnings,
+        additional_verification_payment.subtask_id,
+    )
+    check_warning_payment_value_differs_from_verification_cost_in_settings(
+        warnings,
+        matched_objects,
+        additional_verification_payment,
+    )
+
+
+def check_error_cannot_match_message_with_any_payment(
+    errors: List[Error],
+    matched_objects: List,
+    pending_response: PendingResponse,
+) -> bool:
+    """ Add error if message cannot be matched with any payment."""
+    if len(matched_objects) == 0:
+        errors.append(
+            Error(
+                matching_error=MatchingError.MESSAGE_NOT_MATCHED_WITH_ANY_PAYMENT,
+                message=f'PendingResponse of type {pending_response.response_type} and pk {pending_response.pk} does not have any matching payment.',
+            )
+        )
+        return True
+    return False
+
+
+def check_error_cannot_match_payment_with_any_message(
+    errors: List[Error],
+    matched_objects: List,
+    payment: Any,
+) -> bool:
+    """ Add error if payment cannot be matched with any message."""
+    if len(matched_objects) == 0:
+        errors.append(
+            Error(
+                matching_error=MatchingError.PAYMENT_NOT_MATCHED_WITH_ANY_MESSAGE,
+                message=f'Payment of type {type(payment)} and tx_hash {payment.tx_hash} does not have any matching messages.',
+            )
+        )
+        return True
+    return False
```

Comment:
Why some of these `check_` functions return booleans and others don't?

I think `check_` is not a good name because it's vague and does not really tell you what the function does. Does it return a value? Raise an exception? Just check something and keep quiet about it?

If I were designing it, I'd probably make those functions return the error and warning objects instead of giving it a list that it's supposed to modify.